addons/udes_advanced_packaging/models/stock_picking.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+from collections import defaultdict
+
+from odoo import models, _
+from odoo.exceptions import ValidationError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _should_reserve_full_packages(self):
+        """Method to determine if picking should reserve entire packages"""
```

Comment:
Will this ever be used on a group of pickings with the same type.
Could maybe do the ensure_one on the picking type?

Reply:
What is the advantage of using ensure_one?

Reply:
Currently this is only used in a lambda expression, so the pickings are passed one at a time effectively.

But yeah, no reason to not allow it to be called on a group, so long as they are all the same type.

@caleb-shelton imagine you pass in the following pickings to this method via `self`:

1. A pick picking
2. A goods out picking

Pick picking has full package reservation enabled, but the Goods Out doesn't. What do you return? True for Pick or false for Goods Out? In this scenario we can't give a proper answer, so `ensure_one()` would throw an error as we are using this method wrong.

The following should do the trick I think:

`self.picking_type_id.ensure_one()` - you don't always need to use mapped in Odoo 14, but if it errors you can do `self.mapped("picking_type_id").ensure_one()'.

Reply:
Fixed on latest commit